packages/playground/src/utils/nodeSelector.ts

```diff
@@ -324,21 +324,35 @@ export function isNodeValid(
 }
 
 export async function selectValidNode(
+  gridStore: ReturnType<typeof useGrid>,
   getFarm: GetFarmFn,
   nodes: NodeInfo[],
   selectedMachines: SelectedMachine[],
   filters: FilterOptions,
   oldSelectedNodeId?: number,
   nodesLock?: AwaitLock,
 ): Promise<NodeInfo | void> {
-  let locked = true;
-  if (nodesLock && !nodesLock.acquired) {
-    locked = false;
-    await nodesLock.acquireAsync();
+  const locked = true;
+
+  const rentedNodes = nodes.filter(n => {
+    return n.rentedByTwinId === gridStore.grid.twinId;
+  });
+  let rentedNode;
+  for (const node of rentedNodes) {
+    if (node.rentedByTwinId === gridStore.grid.twinId) {
+      const contractInfo = await gridStore.grid.contracts.get({
+        id: node.rentContractId,
+      });
+
+      if (!contractInfo.state.gracePeriod) {
+        rentedNode = node;
+        break;
+      }
+    }
   }
 
-  if (oldSelectedNodeId) {
-    const node = nodes.find(n => n.nodeId === oldSelectedNodeId);
+  if (oldSelectedNodeId || rentedNode) {
+    const node = rentedNode || nodes.find(n => n.nodeId === oldSelectedNodeId);
```

Comment:
we need to keep the old selected node if it exists.

 ```suggestion
    const node =  nodes.find(n => n.nodeId === oldSelectedNodeId) || rentedNode ;
```

Reply:
I remember in Th BAM we agreed on always choosing the rented node, @AhmedHanafy725 can you confirm that we'll choose the old selected node even if there's a rented node?

Reply:
yes we should select the old selected node